mitosheet/mitosheet/code_chunks/add_column_set_formula_code_chunk.py

```diff
@@ -5,68 +5,69 @@
 # Distributed under the terms of the GPL License.
 
 from copy import copy
-from typing import List, Optional
+from typing import Any, Dict, List, Optional, Tuple
 
 from mitosheet.code_chunks.code_chunk import CodeChunk
 from mitosheet.code_chunks.no_op_code_chunk import NoOpCodeChunk
 from mitosheet.code_chunks.step_performers.column_steps.delete_column_code_chunk import DeleteColumnsCodeChunk
 from mitosheet.code_chunks.step_performers.column_steps.rename_columns_code_chunk import RenameColumnsCodeChunk
 from mitosheet.parser import parse_formula
+from mitosheet.state import State
 from mitosheet.transpiler.transpile_utils import \
     column_header_to_transpiled_code
+from mitosheet.types import ColumnHeader, ColumnID
 
 
 class AddColumnSetFormulaCodeChunk(CodeChunk):
 
+    def __init__(self, prev_state: State, post_state: State, sheet_index: int, column_id: ColumnID, column_header: ColumnHeader, column_header_index: int):
+        super().__init__(prev_state, post_state)
+        self.sheet_index = sheet_index
+        self.column_id = column_id
+        self.column_header = column_header
+        self.column_header_index = column_header_index
+
+        self.df_name = self.post_state.df_names[self.sheet_index]
```

Comment:
So my understanding is that: we used to pass an untyped `param` argument to the code chunks. Then in the code chunk when we wanted to use the params, we needed to get the param from the untyped param, so we used a `get_param` function which just returned the param if it existed and if for some reason it did not, it returned None. 

Essentially, we relied on the type system from the frontend to ensure that the params we expected to be defined for each step were. And if for some reason it wasn't using the `get_param` function was a bit safer. 

Now, instead of using the untyped params, we are explicitly passing the params to the code chunk and setting those params as data fields in the class. For now, this doesn't do a ton for us, but in the future, we can define properties for these data fields like "this param cannot change". 

Alternatively to the change that we just made, we could strongly type our step performer params and then pass those to code chunks just as strongly typed param objects. To my understanding that would give us a very similar benefit to what we have here, but would also save us a ton of lines of code where we have to pass the params one by one to the code chunks and then set those params as data fields. 

With strongly typed params, we could also accomplish things like "this param cannot change" through the type system or if we want to create the classes still, we could unpack the params and create the data fields without having to use a line of code for each param.